packages/contracts/contracts/L2/predeploys/OVM_L1BlockNumber.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import { iOVM_L1BlockNumber } from "./iOVM_L1BlockNumber.sol";
+import { L1Block } from "../L1Block.sol";
+
```

Comment:
Technically the current implementation of this contract will return the L1 block number with just a call to the contract, you don't need to abide by the interface.
I think its unlikely anybody is using the contract in this way, but we may want to have a fallback function that does the same thing as `getL1BlockNumber`, see https://github.com/ethereum-optimism/optimism/blob/2e0b633004b14edd14d9c01fd5119b58680a7dfa/packages/contracts/tasks/take-dump.ts#L109

Reply:
How would that fallback function work? `fallback` can only return `bytes memory`, which is a different API from the one currently in place.

Reply:
We would have to use some assembly to do some type coercion. I think its super unlikely that anybody depends on that API but if we want to be pedantic then we may have to support it. The other problem is now that its written in solidity, it'll use a lot more gas than previously - there will be a jumptable for functions, checks on `msg.value == 0` that solidity does automatically. Unlikely that anybody is relying on that extremely low cost of the raw bytecode, but still mentioning another difference. It will be impossible to get around some increases gas costs since we are going from a simple "push a value from the evm context to the stack" to a "call another contract and load the return value into memory" model.

Its extremely unlikely anybody is using the fallback based API